src/prefect/utilities/callables.py

```diff
@@ -27,7 +27,10 @@ def get_call_parameters(
     Raises a ParameterBindError if the arguments/kwargs are not valid for the function
     """
     try:
-        bound_signature = inspect.signature(fn).bind(*call_args, **call_kwargs)
+        signature_fn = inspect.signature(fn)
+        if "self" in signature_fn.parameters:
```

Comment:
We'll want to do a more robust check. Probably `inspect.ismethod`.

I think we'll also want to move enforcement for this to the `Task.__init__` method instead. That way, a user will get an error when they define the task rather than at runtime.